### DISCUSSION TITLE
Unexpected output dimension when using DynUNet

### DISCUSSION THREAD

**Discussion by lidialuq** [Project-MONAI/MONAI#3437] [Q&A]
I'm trying to use DynUNet on the BraTS dataset. The inputs are of size [1, 4, 128, 128, 128], and I expect the output size to be [1, 3, 128, 128, 128] (batch size of 1, 3 labels) but instead I get [1, 4, 3, 128, 128, 128]. It seems to give a label for each input channel. Any pointers as to what I'm doing wrong would be much appreciated.

Here's how I'm initializing the network: 
```
model = DynUNet(
    spatial_dims=3,
    in_channels=4,
    out_channels=3,
    kernel_size= [[3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3], [3, 3, 3]],
    strides= [[1, 1, 1], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2]],
    upsample_kernel_size= [[2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2], [2, 2, 2]],
    norm_name="instance",
    deep_supervision=True,
    deep_supr_num=3,
)
```
P.S.: I'm not sure if this is the place to ask beginner questions such as this, if not, is there a more appropriate forum?

**Comment by Nic-Ma** [comment]
Hi @lidialuq ,

I think that because it also returns the feature maps:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/networks/nets/dynunet.py#L103

Thanks.
